src/Core/Checkout/Promotion/DataAbstractionLayer/PromotionRedemptionUpdater.php

```diff
@@ -104,23 +119,31 @@ public function update(array $ids, Context $context): void
         }
     }
 
-    public function orderPlaced(CheckoutOrderPlacedEvent $event): void
+    public function orderPlaced(CheckoutOrderPlacedEvent|EntityWrittenEvent $event): void
     {
-        $lineItems = $event->getOrder()->getLineItems();
-        $customer = $event->getOrder()->getOrderCustomer();
+        if ($event instanceof EntityWrittenEvent) {
+            if ($event->getEntityName() !== OrderDefinition::ENTITY_NAME) {
+                return;
+            }
+
+            $criteria = new Criteria($event->getIds());
+            $criteria->addAssociations(['lineItems', 'orderCustomer']);
+
+            $order = $this->orderRepository->search($criteria, $event->getContext())->getEntities()->first();
+            $promotions = $order->getLineItems()->filterByType(PromotionProcessor::LINE_ITEM_TYPE);
+            $customer = $order->getOrderCustomer();
+        } else {
+            $promotions = $event->getOrder()->getLineItems()->filterByType(PromotionProcessor::LINE_ITEM_TYPE);
+            $customer = $event->getOrder()->getOrderCustomer();
+        }
```

Comment:
can't you do two different methods calling one private one afterwards? Would look cleaner and make the method shorter